src/Simulation/EntryPointDriver/Azure/Azure.cs

```diff
@@ -418,7 +418,9 @@ private static void LogIfVerbose(GenerateAzurePayloadSettings settings, string m
             null => null,
             NoOpQirSubmitter.Target => new NoOpQirSubmitter(),
             NoOpSubmitter.Target => new NoOpSubmitter(),
-            _ => SubmitterFactory.QirSubmitter(settings.Target, settings.CreateWorkspace(), settings.Storage)
+            _ => (settings.TargetCapability != null) 
+                ? SubmitterFactory.QirSubmitter(settings.Target, settings.CreateWorkspace(), settings.Storage) 
```

Comment:
I think we should also add the handling for `TargetCapability` to the QIR submitter before merging this to main. `TargetCapability` doesn't do anything until then, and we'd need another PR to qsharp-runtime to change that. Is there a reason this should be merged before the submitter is updated?

Reply:
The changes to the C# client were scheduled as a separate DevOps task, but I added them now, to the internal repo. Please see there.